src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,158 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
+
+function filterPeople(
+  people: Person[],
+  query: string,
+  centuriesSelected: string[],
+  sexSelected: string | null,
+) {
+  return people
+    .filter(person => {
+      if (!sexSelected || sexSelected === 'all') {
+        return true;
+      }
+
+      return person.sex === sexSelected;
+    })
+    .filter(person => {
+      if (centuriesSelected.length === 0) {
+        return true;
+      }
+
+      const personCentury = Math.floor(person.born / 100) + 1;
+
+      return centuriesSelected.includes(String(personCentury));
+    })
+    .filter(person => {
+      if (!query) {
+        return true;
+      }
+
+      return (
+        person.name.toLowerCase().includes(query.toLowerCase().trim()) ||
+        person.motherName?.toLowerCase().includes(query.toLowerCase().trim()) ||
+        person.fatherName?.toLowerCase().includes(query.toLowerCase().trim())
+      );
+    });
+}
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [peopleUsed, setPeopleUsed] = useState<Person[]>([]);
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sortField = searchParams.get('sort') || '';
+  const sortOrder = searchParams.get('order') || '';
+
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries');
+  const sex = searchParams.get('sex') || null;
+
+  useEffect(() => {
+    getPeople()
+      .then(data => {
+        setPeople(data);
+        setPeopleUsed(data);
+        setError(false);
+      })
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    const filteredArr = filterPeople([...people], query, centuries, sex);
+
+    const sortedPeople = [...filteredArr];
+
+    if (sortField) {
+      sortedPeople.sort((a, b) => {
+        const fieldA = a[sortField as keyof Person];
+        const fieldB = b[sortField as keyof Person];
+
+        if (fieldA && fieldB) {
+          if (fieldA < fieldB) {
+            return sortOrder === 'desc' ? 1 : -1;
+          }
+
+          if (fieldA > fieldB) {
+            return sortOrder === 'desc' ? -1 : 1;
+          }
+        }
+
+        return 0;
+      });
+    }
+
+    if (JSON.stringify(sortedPeople) !== JSON.stringify(peopleUsed)) {
+      setPeopleUsed(sortedPeople);
+    }
+  }, [sortField, sortOrder, sex, query, centuries]);
```

Comment:
check this eslint error,  if you don't need fix it - disabled eslint for this case